### ISSUE TITLE
Grouping a single Quote block transforms it (instead of grouping it)

### ISSUE THREAD

**Issue by bacoords** [WordPress/gutenberg#58042]
### Description

When I want to put a Quote block inside of a group, I select "Group" from the dropdown it _transforms_ the block into a group, rather than groups the block. Feels like the expected behavior (based on almost all other blocks) is to group the block, not transform it.

### Step-by-step reproduction instructions

1. Open the block editor and add a Quote.
2. Select the Quote
3. Select the Group option from the dropdown

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/6867360/4e60f35d-69a3-4600-a364-f7d26c2b9287

### Environment info

- Tested in WP 6.4, with AND without Gutenberg (17.5.0)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
I can replicate with WP 6.4.2, TT4, and GB 17.5.1.

https://github.com/WordPress/gutenberg/assets/26996883/897f10c0-f425-43e2-970a-64659406df84

If I turn off GB 17.5.1, I can still replicate with just 6.4.2. Adding to 6.5 board and polish board as needs dev in case we can get a fix in in time.

**Comment by ntsekouras** [comment]
This is intended right now and the responsible code is [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/quote/transforms.js#L126). 

Do you think the most common expectation for this use case is to wrap a Group around though? Personally I think it's better this way, but it's totally subjective. If we end up wanting the suggested behavior, we'd just need to remove the linked transform.

**Comment by bacoords** [comment]
I feel like the intended behavior would make sense had I chosen Group from the "Transform to" menu: 
<img width="334" alt="Screenshot 2024-01-25 at 6 39 51 AM" src="https://github.com/WordPress/gutenberg/assets/6867360/d6e7f4e5-fc59-47ef-94fc-1c389791908a">

But when selecting "Group" from the dropdown, it feels like the behavior shouldn't be to transform the block, it should be to put it inside of a group.

It would be helpful to know what other core blocks actually transform the block type when you select Group from the dropdown (not from the "Transform to" menu) to compare.

**Comment by annezazu** [comment]
>But when selecting "Group" from the dropdown, it feels like the behavior shouldn't be to transform the block, it should be to put it inside of a group.

I would agree with that. It's clear someone is taking an action to add a Group around it rather than transform from the three dot menu.

**Comment by ntsekouras** [comment]
The issue here is that the two buttons(from transforms menu and from the block settings menu) are using the same mechanism to transform. What this means is we don't have an existing way to differentiate the handling between them. 

Is this what we should aim for and explore a solution, or you think it's fine if we wrap the Quote in Group in both cases?

**Comment by bacoords** [comment]
> The issue here is that the two buttons(from transforms menu and from the block settings menu) are using the same mechanism to transform. What this means is we don't have an existing way to differentiate the handling between them.

That's an interesting point. In fact, the only block where the "Transform to Group" action removes the original block (and I've tested almost every block, including the Pull Quote, Media/Text, Gallery, etc) that I can find is the Cover block. In every other block I've tested (WP 6.4.2) the Transform to Group action places the block inside of a group but keeps the original.

**Comment by bacoords** [comment]
I should clarify - for the cover block, the "Transform to Group" action transforms the cover into a group, but selecting "Group" from the kebab dropdown menu wraps it in a group - which I think is essentially the same behavior that the Quote block _should_ have.

**Comment by ntsekouras** [comment]
> for the cover block, the "Transform to Group" action transforms the cover into a group, but selecting "Group" from the kebab dropdown menu wraps it in a group

That surprised me when I read it and checked again, but doesn't seem to be the case. It's true that for core blocks the ones that replace the block are `Quote and Cover`, but both buttons(transform menu and block settings menu) perform the same action, which is change the block type and not wrap.

So, my above comment still stands whether we would like different behavior, which would call for a new exploration and how useful it would be for other blocks too.

**Comment by bacoords** [comment]
>  It's true that for core blocks the ones that replace the block are Quote and Cover, but both buttons(transform menu and block settings menu) perform the same action, which is change the block type and not wrap.

Well that's interesting: So if your cover block is empty/untouched, it _transforms_ into a group, but if your cover block has a background image assigned already, it _wraps_ the block (whether you use the transform menu or the block settings menu).

In either case I think the goal of raising this issue was to point out that the UI on this is very inconsistent and that the simplest approach (simple from the user perspective, not the dev perspective) could be: 

-  the Group option in the "Transform" menu would transform the block
-  the Group option in the block settings menu would wrap the block 

This would be applied across all blocks consistently and will be even more critical if we see #46972 merged. I think that's the behavior implied by the UI and would be happy to help with or test a PR on this if other agree that that makes sense. 

Would be interested in feedback from a UI perspective as well (@richtabor)

**Comment by annezazu** [comment]
At this stage, we're outside the ability to quickly make a decision here for 6.5. I'm going to punt it from the release as a result.

**Comment by carolinan** [comment]
This was also reported on Trac in December 2023: https://core.trac.wordpress.org/ticket/60124

>When using the Block Editor and attempting to 'group' an existing Quote block, the 'Quote' block gets converted to a 'Paragraph' block.
My expectation is that the Quote block would be grouped inside of a Group block, but instead it seems to convert the Quote block into a Group block and then has the quote content as a paragraph block within.

**Comment by colorful-tones** [comment]
Hi folks,
We are only one week away from the Beta 1 cut-off date for [WordPress 6.6](https://make.wordpress.org/core/6-6/). This issue hasn’t seen any movement in a while, so we (the editor triage leads of the 6.6 release) have decided to remove it from the [WordPress 6.6 Editor Tasks project board](https://github.com/orgs/WordPress/projects/179/views/1).
